EasyPost.Tests/BetaFeaturesTests/ParametersTests/ParametersTest.cs

```diff
@@ -0,0 +1,152 @@
+            Assert.True(addressData.ContainsKey("street2"));
+            Assert.True(addressData.ContainsKey("city"));
+            Assert.True(addressData.ContainsKey("state"));
+            Assert.True(addressData.ContainsKey("zip"));
+            Assert.True(addressData.ContainsKey("country"));
+            Assert.True(addressData.ContainsKey("residential"));
+
+            // Check that the "street1" key was set with the correct value
+            Assert.NotNull(addressData["street1"]);
+            Assert.Equal(street, addressData["street1"]);
+        }
+
+        [Fact]
+        [Testing.Parameters]
+        public void TestParametersToDictionaryWithSubDictionary()
+        {
+            const string streetA = "388 Townsend St";
+            const string streetB = "123 Main St";
+
+            Address preExistingAddressObject = new Address
+            {
+                Street1 = streetA,
+            };
+
+            Addresses.Create newAddressCreationParameters = new Addresses.Create
+            {
+                Street1 = streetB,
+            };
+
+            // Users can pass in an existing Address object as the "ToAddress" parameter
+            var parameters = new Shipments.Create
+            {
+                IsReturn = false,
+                ToAddress = preExistingAddressObject,
+            };
+
+            // No errors here confirm that the dictionary was serialized to the expected schema
+            var dictionary = parameters.ToDictionary();
+            var shipmentData = dictionary["shipment"] as Dictionary<string, object>;
+            var addressData = shipmentData["to_address"] as Dictionary<string, object>;
+
+            // The value of "street1" should be the value of "streetA" via the Address object
+            Assert.Equal(streetA, addressData["street1"]);
+
+            // Users can also pass in an Address.Create parameter object as the "ToAddress" parameter
+            parameters = new Shipments.Create
+            {
+                IsReturn = false,
+                ToAddress = newAddressCreationParameters,
+            };
+
+            // No errors here confirm that the dictionary was serialized to the expected schema
+            dictionary = parameters.ToDictionary();
+            shipmentData = dictionary["shipment"] as Dictionary<string, object>;
+            addressData = shipmentData["to_address"] as Dictionary<string, object>;
+
+            // The value of "street1" should be the value of "streetB" via the Address.Create parameter object
+            Assert.Equal(streetB, addressData["street1"]);
+        }
+
+        [Fact]
+        [Testing.Logic]
+        public void TestTopLevelVersusNestedParameters()
+        {
+            // This test proves that we can reuse the Address.Create parameter object,
+            // with its serialization logic adapting to whether it is
+            // a top-level parameter object or a nested parameter object within another parameter object.
```

Comment:
This is exactly the kind of stuff I'd hope each test have in a docstring, can we move it to a docstring instead of in the function?